save-osv/src/main/kotlin/com/saveourtool/save/osv/controllers/OsvController.kt

```diff
@@ -0,0 +1,87 @@
+package com.saveourtool.save.osv.controllers
+
+import com.saveourtool.save.configs.ApiSwaggerSupport
+import com.saveourtool.save.configs.RequiresAuthorizationSourceHeader
+import com.saveourtool.save.entities.vulnerability.*
+import com.saveourtool.save.osv.service.OsvService
+import com.saveourtool.save.utils.*
+import com.saveourtool.save.v1
+
+import org.springframework.http.MediaType
+import org.springframework.http.ResponseEntity
+import org.springframework.http.codec.multipart.FilePart
+import org.springframework.security.core.Authentication
+import org.springframework.web.bind.annotation.*
+import reactor.core.publisher.Flux
+import reactor.core.publisher.Mono
+
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+
+/**
+ * Rest controller for OSVs
+ */
+@ApiSwaggerSupport
+@RestController
+@RequestMapping("/api/$v1/osv")
+class OsvController(
+    private val osvService: OsvService,
+) {
+    /**
+     * @param name vulnerability name in save db
+     * @return content of OSV
+     */
+    @GetMapping(path = ["/get-by-save-name/{name}"], produces = [MediaType.APPLICATION_JSON_VALUE])
```

Comment:
may be change an API for the future:
/get-by-id/save/{name}

It will be extended with:
/get-by-id/cve/{name}

Reply:
actually, I added this method is for testing purpose -- to show that the link does exist.
I think we should return by a single ID (primary id).

Right now we create an entity `com.saveourtool.save.entities.vulnerabilities.Vulnerability` with generated name (id) and `vulnerabilityIdentifier` from original OSV file. We should not generate save's vulnerability id for such uploads (use id from OSV file).
We should generate ID only when we is original for vulnerability.

I can support using the original ID already in this PR

Reply:
supported and refactored these methods